core/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/HistogramAggregationBuilder.java

```diff
@@ -185,17 +186,32 @@ public HistogramAggregationBuilder extendedBounds(double minBound, double maxBou
     }
 
     /** Return the order to use to sort buckets of this histogram. */
-    public Histogram.Order order() {
+    public BucketOrder order() {
         return order;
     }
 
     /** Set a new order on this builder and return the builder so that calls
      *  can be chained. */
-    public HistogramAggregationBuilder order(Histogram.Order order) {
+    public HistogramAggregationBuilder order(BucketOrder order) {
         if (order == null) {
             throw new IllegalArgumentException("[order] must not be null: [" + name + "]");
         }
-        this.order = (InternalOrder) order;
+        if(order instanceof CompoundOrder || InternalOrder.isKeyOrder(order)) {
+            this.order = order;
+        } else {
+            this.order = BucketOrder.compound(order);
```

Comment:
We may need to add a tie-breaker sort (`_key` asc) to prevent non-deterministic ordering. See #23613